subprojects/docs/src/docs/release/notes.md

```diff
@@ -258,9 +258,12 @@ Related issues:
 
 Gradle 7.6 supports compiling, testing and running on Java 19.
 
-#### TODO: Introduced support for map notation in JVM `dependencies` block
+#### Introduced strongly-typed JVM test suites `dependencies` block
 
-[Issue](https://github.com/gradle/gradle/issues/19192)
+The test suites' `dependencies` block is now strongly-typed. This means that the type of added dependencies is now
+strongly-typed. For example, using a `String` will provide an `ExternalModuleDependency`, whereas using a `FileCollection` will
+provide a `FileCollectionDependency`. This allows Java and Kotlin to properly configure all types of dependencies, and improves
+IDE support for Groovy.
```

Comment:
```suggestion
The test suite `dependencies` block is now [strongly typed](https://en.wikipedia.org/wiki/Strong_and_weak_typing).

Previously, you could only dependencies in a test suite with strings:

```kotlin
testing {
    suites {
        val test by getting(JvmTestSuite::class) {
            useJUnitJupiter()
            dependencies {
                implementation("org.junit.jupiter:junit-jupiter:5.8.2")
                runtimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.2")
            }
        }
    }
}
```

Now, you can use map notation to define a dependency's group, name,
and version separately:

```kotlin
testing {
    suites {
        val test by getting(JvmTestSuite::class) {
            useJUnitJupiter()
            dependencies {
                implementation(group = "org.junit.jupiter", name = "junit-jupiter", version = "5.8.2")
                runtimeOnly(group = "org.junit.jupiter", name = "junit-jupiter-engine", version = "5.8.2")
            }
        }
    }
}
```

For example, using a `String` provides an `ExternalModuleDependency`.
Using a `FileCollection` provides a `FileCollectionDependency`.
This allows Java and Kotlin to properly configure all types of dependencies
and improves Groovy IDE support.
```

Nitpick: Added link to "strongly typed" definition.
Question: Would it better to use the term "type safe" here? I'm not sure which is more correct based on the PR.
Suggestion: Add code blocks to demonstrate the change.

Also trimmed some words when possible.

Reply:
In my opinion "strongly typed" and "type safe" are basically the same thing. I'd prefer to keep the description consistent between all the sources, e.g. #21209 #20868.

Also, I would like to note that the Kotlin syntax is not "map notation", it does not involve a Map at any time. It's just [named arguments](https://kotlinlang.org/docs/functions.html#named-arguments). I wish to avoid confusion with (a) the Groovy syntax that does use a Map, and (b) the old methods that used to accept a Map, which are no longer present. All three of these are separate parts, though they appear similar.